### PR TITLE
Add new python interpreters to test suite

### PR DESCRIPTION
**Pull request by thunderboltsid**
Same as #9552 but for base v1.1.x

### CODE REVIEW

File: tools/run_tests/run_tests.py [hunk 1]
```diff
@@ -712,7 +713,7 @@ def test_specs(self):
         runtime_cmd = ['mono']
 
     specs = []
-    for assembly in tests_by_assembly.iterkeys():
```
Comment (nathanielmanistaatgoogle) [inline tools/run_tests/run_tests.py]: Just drop the `.iterkeys()`entirely and go with `for assembly in tests_by_assembly:`?

Comment (thunderboltsid) [inline tools/run_tests/run_tests.py]: this one escaped my eyes iter(tests_by_assemble.keys()) should yield the same results in both

Comment (nathanielmanistaatgoogle) [inline tools/run_tests/run_tests.py]: We use `six` for all 2-and-3-together compatibility requirements, so `six.iterkeys(tests_by_assembly)` in that case?

**Comment by jtattermusch** [comment]
Thanks for the PR, but there are several issues:
-- I believe the v1.1.x branch is locked now (we cut v1.2.x release branch yesterday and no more changes are being done to v1.1.x)
-- Why does this PR even go against v1.1.x  (wouldn't it be better to create against master)
-- I see a lot of python code style / formatting improvements in this PR. That's fine but all such changes should strictly come in a separate PR unless you are actually changing a specific line of code for other reasons. With all the formatting changes, it really hard to see the real diff in functionality. Also, no formatting changes should be done in a release branch because that just doesn't make sense - improving code style only belongs to the master branch

For the reasons above, I recommend closing this PR and starting a fresh one against master (unless some fixes are needed in v1.2.x and in that case only the bugfixes should go there).

**Comment by nathanielmanistaatgoogle** [comment]
@jtattermusch: this feature has been in development since November (before it was clear that we were going to increase our release cadence). It was targeted at `v1.1.x` because that's what made sense at the time that this particular pull request was opened and I asked for it to be targeted at `v1.1.x`. Since GitHub now lets us retarget pull requests, I've retargeted it at `master`; `master`'ll probably never be out of date.

@thunderboltsid: I agree with @jtattermusch about breaking out the stylistic bits. I imagine you're greatly frustrated by how long this has hung out in development; I'm frustrated too but I think breaking it into smaller pieces will help it find sooner resolution. Defining new environment variables? Maybe its own pull request. Making `run_tests/**.py` compatible with Python 3? Definitely its own pull request. Lastly, I apologize for erring in my estimation of when and on what branch this would land.

**Comment by thunderboltsid** [comment]
This PR is now headed for master instead of v1.1.x as requested. My editor is generally set to follow the boy-scout principle and keeps cleaning stuff on it's own. Apologies for that. I have now reverted those changes manually. The reason why making test scripts python3.x compatible is part of this PR is so that the version of python that is being tested can also test itself.

**Comment by nathanielmanistaatgoogle** [comment]
At some point I intend to get [`yapf`](https://github.com/grpc/grpc/issues/262) and [`pylint`](https://github.com/grpc/grpc/issues/261) coverage extended to everything in `tools` so that the boy scouting of that code is included in our test suite in the same manner as it is for our `src/python` code.

**Comment by jtattermusch** [comment]
@thunderboltsid  understood and thanks!

**Pull request closed without merging**